File: 2021/day1/day1.go
package main

import (
	"fmt"
    "advent-of-code/tools"
)

func part1(data []int) int{
    inc := 0
    for i := 1; i < len(data); i++{
        if data[i] > data[i-1]{
            inc++
        }
    }
    return inc
}

func part2(data []int) int{
    inc := 0
    for i:=0; i < len(data)-3; i++{
        if data[i+3] > data[i]{
            inc ++
        }
    }
    return inc

}

func main() {
    data := tools.Read("day1a.txt")
    fmt.Println(part1(data))
    fmt.Println(part2(data))
}

File: 2021/day1/day1.py
def read_data(filename):
    with open(filename, "r") as fp:
        data = fp.read().strip().split("\n")
    return list(map(int, data))


t_data = [
    199,
    200,
    208,
    210,
    200,
    207,
    240,
    269,
    260,
    263,
]
data = read_data("day1a.txt")

# PART 1
def part1(data):
    inc = sum(1 for i in range(1, len(data)) if data[i] > data[i - 1])
    return inc


assert part1(t_data) == 7
print(part1(data))

# PART 2
def part2(data):
    inc = sum(data[i+3] > data[i] for i in range(0, len(data)-3))
    return inc


assert part2(t_data) == 5
print(part2(data))
